dependencies/getSMAIRMatrix.m

```diff
@@ -79,68 +82,85 @@
+    % the rendered BRIRs to start with a positive peak (which seems
+    % reasonable). Without the minus, the resulting BRIRs are inverted.
+    bnAll = -sphModalCoeffs(simulationOrder, 2*pi*f/C * params.smaRadius, ...
+        params.arrayType, params.dirCoeff).';
 
-    pMics = zeros(numMics, numShsSimulation, nfft);
-
-    for ii = 1:numFreqs
-        Bn = diag(sh_repToOrder(bnAll(ii,:).').');
-        pMics(:,:,ii) = YMics.' * Bn;
-
-        if ii > 1 && ii < numFreqs
-            pMics(:,:,end-ii+2) = YMics.' * conj(Bn);
+    pMics = zeros(numMics, numShsSimulation, nfft, 'like', bnAll);
+    pN = zeros(numShsOut, numShsSimulation, nfft, 'like', bnAll);
+    for k = 1:numPosFreqs
+        Bn = diag(sh_repToOrder(bnAll(:,k)));
+        pMics(:,:,k) = Y_Hi * Bn;
+        if k == numPosFreqs && ~mod(nfft, 2) % is even
+            pMics(:,:,k) = Y_Hi * real(Bn);
+        elseif ~isreal(Y_Hi) && k > 1
+            k_neg = nfft-k+2;
+            pMics(:,:,k_neg) = Y_Hi * conj(Bn);
         end
-    end
 
-    pMics(isnan(pMics)) = 0; % remove singularity for f = 0
-
-    pN = zeros(numShsOut, numShsSimulation, nfft);
-    pinvYMicsLow = pinv(YMicsLow.');
-    for ii = 1:nfft
-        pN(:,:,ii) = pinvYMicsLow * pMics(:,:,ii);
-    end
-
-    % apply radial filtering
-    if (~strcmp(params.radialFilter, 'none'))
-        radFilts = getRadialFilter(params);
-
-        smairMat = zeros(numShsOut, numShsSimulation, nfft);
-        for ii = 1:numFreqs
-            BnTi = diag(sh_repToOrder(radFilts(ii,:).').');
-            smairMat(:,:,ii) = BnTi * pN(:,:,ii);
-
-            if ii > 1 && ii < numFreqs
-                smairMat(:,:,end-ii+2) = conj(BnTi) * pN(:,:,ii);
+        if ~params.returnRawMicSigs
+            pN(:,:,k) = Y_Lo_pinv * pMics(:,:,k);
+            if k == numPosFreqs && ~mod(nfft, 2) % is even
+                pN(:,:,k) = Y_Lo_pinv * real(pMics(:,:,k));
+            elseif ~isreal(Y_Hi) && k > 1
+                pN(:,:,k_neg) = Y_Lo_pinv * pMics(:,:,k_neg);
             end
         end
-
-        smairMat(isnan(smairMat)) = 0;
-    else
-        smairMat = pN;
     end
 
     if params.returnRawMicSigs
         smairMat = pMics;
+    else
+        smairMat = pN;
+
+        if ~strcmpi(params.radialFilter, 'none')
+            % apply radial filtering
+            radFilts = getRadialFilter(params).';
+            for k = 1:numPosFreqs
+                BnTi = diag(sh_repToOrder(radFilts(:,k)));
+                smairMat(:,:,k) = BnTi * smairMat(:,:,k);
+                if k == numPosFreqs && ~mod(nfft, 2) % is even
+                    smairMat(:,:,k) = real(BnTi) * smairMat(:,:,k);
+                elseif ~isreal(Y_Hi) && k > 1
+                    k_neg = nfft-k+2;
+                    smairMat(:,:,k_neg) = conj(BnTi) * smairMat(:,:,k_neg);
+                end
+            end
+        end
     end
-    
\ No newline at end of file
+
+    % TODO: resolve system-dependent differences in resulting eMagLS and eMagLS2 filters
```

Comment:
This can be used to export the data on different systems and compare it directly. My data is also contained in 6a2fc2bdd6a823ff977f4fefa12279c32a900aab.